dev/helpers/mssql/migrate_migrations.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# There seems to be [a bug with docker-compose](https://github.com/docker/compose/issues/4076#issuecomment-324932294) 
+# where it takes ~40ms to connect to the terminal output of the container, so stuff logged to the terminal in this time is lost.
+# The best workaround seems to be adding tiny delay like so:
+sleep 0.1;
+
+MIGRATE_DIRECTORY="/mnt/migrator/DbScripts"
+LAST_MIGRATION_FILE="/mnt/data/last_migration"
+SERVER='mssql'
+DATABASE="vault_dev"
+USER="SA"
+PASSWD=$MSSQL_PASSWORD
+
+while getopts "s" arg; do
+  case $arg in
+    s)
+      echo "Running for self-host environment"
+      LAST_MIGRATION_FILE="/mnt/data/last_self_host_migration"
+      DATABASE="vault_dev_self_host"
+      ;;
+  esac
+done
+
+if [ ! -f "$LAST_MIGRATION_FILE" ]; then
+  echo "No migration file, nothing to migrate to a database store"
+  exit 1
+else
+  LAST_MIGRATION=$(cat $LAST_MIGRATION_FILE)
+  rm $LAST_MIGRATION_FILE
+fi
+
+[ -z "$LAST_MIGRATION" ]
+PERFORM_MIGRATION=$?
+
+# Create database if it does not already exist
+QUERY="IF NOT EXISTS (SELECT * FROM sys.databases WHERE name = 'migrations_$DATABASE')
```

Comment:
Why a separate database and not just a special table in the same database like EF does it?

Reply:
🤷 not a super good reason I guess, but I was thinking we use a new db just to not limit the names available for tables for product.